specification/marketplacecatalog/resource-manager/Search/stable/2022-02-02/search.json

```diff
@@ -0,0 +1,785 @@
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "FacetsItemDto": {
+      "type": "object",
+      "properties": {
+        "facetName": {
+          "type": "string"
+        },
+        "facetValues": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FacetValueDto"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "FacetValueDto": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "string"
+        },
+        "count": {
+          "type": "integer",
+          "format": "int64"
+        }
+      }
+    },
+    "SearchResponseDto": {
+      "type": "object",
+      "properties": {
+        "facets": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FacetsItemDto"
+          },
+          "x-ms-identifiers": []
+        },
+        "results": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SearchItemDto"
+          },
+          "x-ms-identifiers": [
+            "legacyId"
+          ]
+        },
+        "totalCount": {
+          "type": "integer",
+          "format": "int64"
+        }
+      }
+    },
+    "SearchItemDto": {
+      "type": "object",
+      "properties": {
+        "ampAppsPopularity": {
+          "type": "number",
+          "format": "double"
+        },
+        "ampCsPopularity": {
+          "type": "number",
+          "format": "double"
+        },
+        "applicableProducts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "appSourceAppsPopularity": {
+          "type": "number",
+          "format": "double"
+        },
+        "appSourceCategories": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "appSourceCsPopularity": {
+          "type": "number",
+          "format": "double"
+        },
+        "appSourceIndustries": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "azureCategories": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "azurePortalCategories": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "azureIndustries": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "bigId": {
+          "type": "string"
+        },
+        "cspStates": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "determinedStorefronts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "displayName": {
+          "type": "string"
+        },
+        "isAzureBenefitEligible": {
```

Comment:
>isAzureBenefitEligible [](http://example.com/codeflow?start=9&length=22)

There are a bunch of booleans in this response. ARM recommends string enums over boolean. Generally enums make better properties than booleans. They are more descriptive, flexible and future-proof while being easier for customers to discover, understand and use. String enums allow adding values without a new API version, which looks like it could be a potential benefit for some of these.

Ref: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html?q=boolean#common-issues-found-in-review).

As an enum, this property could be something like:

benefitEligibility: { Default (or None), Azure }.

Reply:
@mentat9 I understand your point but unfortunately this property is based on cosmosDB property that has similar name so I would want to keep the same naming and type for this field so it wont confuse our clients that use both Search API and other API's that get data from our cosmos.